src/Misc/layoutbin/darwin.svc.sh.template

```diff
@@ -63,7 +63,7 @@ function install()
     mkdir -p "${log_path}" || failed "failed to create ${log_path}"
 
     echo Creating ${PLIST_PATH}
-    sed "s/{{User}}/${SUDO_USER:-$USER}/g; s/{{SvcName}}/$SVC_NAME/g; s@{{RunnerRoot}}@${RUNNER_ROOT}@g; s@{{UserHome}}@$HOME@g;" "${TEMPLATE_PATH}" > "${TEMP_PATH}" || failed "failed to create replacement temp file"
```

Comment:
This [syntax](https://pubs.opengroup.org/onlinepubs/009604499/utilities/xcu_chap02.html) indicates to use `SUDO_USER` if it is set, otherwise we should be fallback to `USER`. I think that is the correct behavior here,  Is sudo user not getting set for some reason?

Reply:
No, you are right, it is getting set. We need sudo for the systemd service to run install, but we don't on darwin. The issue here is that the behaviour is confusing to the user issuing the installation of the service.
If let's say I am user `nikola`, and I want to switch to the user `test` to install this service, I would issue the command: `sudo su test`. Now I am a test user, but my `SUDO_USER` would be nikola. When I run `./svc.sh install`, the service would be installed for user `nikola` who issued this command. It would be clearer, since we don't need sudo permissions to enable service on the darwin architecture, to install service as the user issuing the `./svc.sh install`.

To install this service as the user `test`, I would need to issue another sudo command changing the `SUDO_USER`, or to unset the `SUDO_USER`. This behaviour makes it a bit tricky for users who want to issue installation of the service, to install it as another user.

This fix tries to prefer the user invoking the installation, so when I switch to user test, and invoke installation, I would install this service as the user who invoked the service. No the one who ran `sudo su test`. We can do it differently, like in systemd, to pass an argument to install service as another user, but this seemed like a quicker fix, without changing the code that much.